examples/js/loaders/OBJLoader2.js

```diff
@@ -221,7 +220,7 @@ THREE.OBJLoader2 = (function () {
 		}
 
 		Parser.prototype.setDebug = function ( debug ) {
-			this.debug = ( debug == null ) ? this.debug : debug;
+			this.debug = Boolean( debug ) ? debug : this.debug;
```

Comment:
You are correct. I wanted to make things more elegant, but I introduced potential problems, that did not exist before.
I will kill all the Boolean stuff and replace it with correct helper functions inside `OBJLoader2`. These four functions will cover all cases (I wrote some local test to verify). Don't know if I keep `verifyBooleanDefaultFalse` and `verifyBooleanDefaultTrue` as they seem superfluous, but the comment on what happens with non boolean is helpful (at least for me ;-):
```
var Helper = {
	/**
	 * Verifies boolean input. Non-boolean will return false.
	 *
	 * @param input Anything
	 * @returns {boolean}
	 */
	verifyBooleanDefaultFalse: function( input ) {
		return input === true;
	},
	/**
	 * Verifies boolean input. Non-boolean will return true.
	 *
	 * @param input Anything
	 * @returns {boolean}
	 */
	verifyBooleanDefaultTrue: function( input ) {
		return input !== false;
	},
	/**
	 * If given input is null or undefined, false is returned otherwise true.
	 *
	 * @param input
	 * @returns {boolean}
	 */
	isValid: function( input ) {
		return ( input !== null && input !== undefined );
	},
	/**
	 * If given input is null or undefined, the defaultValue is returned otherwise the given input.
	 *
	 * @param input
	 * @param defaultValue
	 * @returns {*}
	 */
	verifyValid: function( input, defaultValue ) {
		return ( input === null || input === undefined ) ? defaultValue : input;
	}

};
```

Then
`this.debug = debug === true ? debug : this.debug;`
could be expressed as follows:
`if ( Helper.verifyBooleanDefaultFalse( debug ) ) this.debug = true;`

Further examples:
`this.streamMeshes = ( streamMeshes === null || streamMeshes === undefined ) ? true : streamMeshes;`
will become:
`this.streamMeshes = Helper.verifyBooleanDefaultTrue( streamMeshes );`

`modelName: Boolean( modelName ) ? modelName : 'none'`
will become:
`modelName: Helper.verifyValid( modelName, 'none' )`

`if ( ! Boolean( this.worker ) )`
will become:
`if ( ! Helper.isValid( this.worker ) )`